src/main/java/com/exadel/discount/filters/JwtFilter.java

```diff
@@ -0,0 +1,105 @@
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.security.web.authentication.preauth.PreAuthenticatedAuthenticationToken;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import static com.exadel.discount.config.JwtVariablesConfig.REFRESH_ROLE;
+
+@Component
+@Setter
+@RequiredArgsConstructor
+public class JwtFilter extends OncePerRequestFilter {
+    private final String AUTHORIZATION_HEADER = "Authorization";
+    private final String BEARER_TYPE_OF_AUTHORIZATION_HEADER = "Bearer ";
+
+    private final UserDetailsServiceImpl userDetailsService;
+
+    private AbstractAuthenticationToken authentication;
+    private String token;
+    private String username;
+    private String role;
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request,
+                                    HttpServletResponse response,
+                                    FilterChain filterChain) throws ServletException, IOException {
+        final String authorizationHeader = request.getHeader(AUTHORIZATION_HEADER);
+
+        if (isNotAuthorized() && hasSuitableAuthorizationHeader(authorizationHeader)) {
+            extractPayload(authorizationHeader);
+
+            if (hasValidSearchKeyForStorage()) {
+                UserDetails userDetails = loadUserDataFromStorageThroughSearchKey();
+
+                if (StringUtils.equals(role, JwtVariablesConfig.getProperty(REFRESH_ROLE))) {
+                    if (JwtUtil.validateToken(token, userDetails)) {
+                        this.authentication = new PreAuthenticatedAuthenticationToken(
+                                userDetails, null, RefreshJwtUtil.getRefreshAuthority());
+                    }
+                } else if (JwtUtil.validateToken(token, userDetails)) {
+                    this.authentication = new UsernamePasswordAuthenticationToken(
+                            userDetails, null, userDetails.getAuthorities());
+
+                }
+                setSecurityContextAuthentication(request, authentication);
+            }
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private boolean isNotAuthorized() {
+        return SecurityContextHolder.getContext().getAuthentication() == null;
+    }
+
+    private boolean hasSuitableAuthorizationHeader(String authorizationHeader) {
+        return
+                StringUtils.isNoneEmpty(authorizationHeader) &&
```

Comment:
StringUtils.isNoneEmpty need to be used for list of strings, not for the individual one